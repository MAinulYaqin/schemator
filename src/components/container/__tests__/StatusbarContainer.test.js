import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";

import ConnectedComponent, { StatusbarContainer } from "../StatusbarContainer";

describe("StatusbarContainer", () => {
  const setup = propOverrides => {
    const props = {
      project: {
        name: "LaravelScheme",
        zoom: 75,
        isModified: false
      },
      ...propOverrides
    };

    const wrapper = shallow(<StatusbarContainer {...props} />);

    return {
      wrapper,
      props
    };
  };

  const setupConnected = () => {
    const mockStore = configureStore();
    const store = mockStore({
      project: {
        name: "LaravelScheme",
        zoom: 75,
        isModified: false
      }
    });

    const wrapper = shallow(<ConnectedComponent store={store} />);

    return {
      wrapper,
      store
    };
  };

  it("should renders with project", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should renders without opened project", () => {
    const { wrapper } = setup({ project: null });

    expect(wrapper).toMatchSnapshot();
  });

  it("should maps state to props", () => {
    const { wrapper, store } = setupConnected();
    const { project } = store.getState();

    expect(wrapper.props()).toMatchObject({ project });
  });
});
