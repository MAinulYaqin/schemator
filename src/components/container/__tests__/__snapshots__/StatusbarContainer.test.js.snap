// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusbarContainer should renders with project 1`] = `
<Statusbar
  isProjectModified={false}
  projectName="LaravelScheme"
  zoom={75}
/>
`;

exports[`StatusbarContainer should renders without opened project 1`] = `
<Statusbar
  zoom={100}
/>
`;
